### ISSUE TITLE
Create Library of RChain

### ISSUE THREAD

**Issue by pmoorman** [rchain/bounties#1004]
## Overview
We're working on a decidedly decentralised effort to craft one central place with "all things RChain".

This initiative has evolved from discussions with @PatrickM727 and @jonathanks (handle?) about an RChain library. The rough idea is that if someone discovers RChain and would like to do a "deep dive" to learn more about it, they should have a reputable source to get that info.

The platform should have the amount and depth of content that makes someone say _"yeah, I feel comfortable to commit to building on this platform!"_

Live work-in-progress:  https://rchain-docs.netlify.com/

## Sponsorship
The DevEd label can provide sponsorship for writing content focused on onboarding and getting developers. In particular, we could use help with the following:

1. Surfacing the most interesting and relevant pieces of content from JIRA/Confluence and developer.rchain.coop, and making it available in an easy-to-read format. (e.g. info on "how does sharding work?" and "how do namespaces relate to each other?" and stuff like that)

2. Writing documentation for Rholang

3. Writing documentation about validator economics, running nodes, etc. 

4. "Getting started" information, to help people quickly find the resources they need to take a next step.

I imagine we'll have to synchronize with @JoshOrndorff about the details & sponsorship amounts once you are ready to commit to a certain scope of work. Both Joshy and Patrick have expressed written support for this project already.

## How to get started
Let me know here what content you'd like to work on. For edit access, reach me here or on Discord (@pmoorman). You can also contribute by compiling information for a relevant page, and providing it here in markdown format (and I'll integrate it in the site).

**Comment by dckc** [comment]
jonathanks = @ottermagically

**Comment by dckc** [comment]
Why doesn't this follow the ISSUE_TEMPLATE task form?

**Comment by pmoorman** [comment]
know what's weird, I can never actually tag @ottermagically. I don't know why it doesn't work, but it doesn't. Only person I'm aware of that doesn't work, everyone else is fine!

> Why doesn't this follow the ISSUE_TEMPLATE task form?

I thought it was more "guideline" rather than a requirement. I try to strip out anything that I deem non-essential. Are there any parts I should add back in?

**Comment by pmoorman** [comment]
@JoshOrndorff if you could jump in here and "officially" say _"this looks good to me"_, that would be awesome 💯 👍

**Comment by dckc** [comment]
> I thought it was more "guideline" rather than a requirement.

If you just want to chit-chat, then we can add the **Discussion** label. But if you're seeking a bounty reward, it needs the SMART stuff:

 - [ ] Estimated Budget of Task:
 - [ ] Estimated Timeline Required to Complete the Task
 - [ ] How will we measure completion?

**Comment by pmoorman** [comment]
If you all don't mind, I'll keep track of regular progress here. It's not funded, but I'd like to work in the open, anyway.

### Yesterday

1. Added a scaffold for a [team page](https://rchain-docs.netlify.com/ecosystem/team.html) (@TrenchFloat maybe you feel like sorting out the content, based on this template?)

2. Added a page for [conferences & meetups](https://rchain-docs.netlify.com/ecosystem/conferences.html) (same here, the content needs some more work)

3. Fixed styling, so we have cool "tips" boxes now, in RChain-branded colors (see conference page).   Also added a favicon.

### Today

4. Added basic content to the [research page](https://rchain-docs.netlify.com/platform/research.html)

Also spoke with other stakeholders about how this piece of work can relate to their efforts (@PatrickM727 for marketing, @ian-bloom @barneycinnamon and Karen for governance, Ned from RChain Solutions)

**Comment by azazime** [comment]
Is this some kind of a joke or the restructuring was not what we thought... A library??? And for that amount???

**Comment by dckc** [comment]
I'm not sure what you mean by "the restructuring"; help me out?

p.s. The **needs-SMART-objective** label is not for objectives that you disagree with; it's for issues lacking an objective altogether.

**Comment by pmoorman** [comment]
@azazime I don't quite understand your comment. You seem to be sarcastic, but it's hard to pick up where you're going at. Please clarify what your concerns are

**Comment by azazime** [comment]
Well its not a very terrible idea but I feel there's already so many things that is more like a library but is been under-used like RUB, Divydao, the community forum, Rchain official blog and so on. The idea is not the worst in the world but for that amount I can hardly see the "value for money"

**Comment by dckc** [comment]
Fair point. We're still brainstorming to some extent. The budget is just
something to start a conversation. Evidently it worked :)

**Comment by pmoorman** [comment]
> I feel there's already so many things that is more like a library but is been under-used like RUB, Divydao, the community forum, Rchain official blog and so on.

**Fair point, and I agree.** Part of what we're learning is that just building platforms without a distribution / engagement strategy, isn't working. When we commit to building something, we should also commit to marketing it, inside and outside of the community.

Let me use this opportunity to _make a start_ with addressing that concern. Here's how I'm planning to drive traffic to this platform:

1. I have been talking to close to 20 people about this initiative, including Patrick, Kevin W. and Joshy from marketing to devs / technical people that can contribute to the archdoc. The feedback has been overwhelmingly positive, and engaging these people early helps drive engagement. I am working actively on bringing more collaborators on board.

2. Patrick has already expressed support to host this on library.rchain.coop. That specific subdomain might still change, but this will be hosted as part of the official RChain experience.

3. I am engaged with the main website team, and we will sync it into the experience there. I'm also in touch with Derek Beres who manages the email marketing efforts (amongst other things), and I think we can plug this platform into the onboarding there.

4. Once the platform is live, we would do well to work on a basic backlink / SEO strategy, to make this platform pop up in Google. I have experience doing this, but in all cases it's still lots of dumb and boring work to build a quality backlink profile.

5. On the long run, quality content will reign. If we get the majority of stakeholders to communicate evergreen content here, that'll be a sustainable long-term strategy. We achieve this through open and straight-forward write access, so many people can edit and contribute in parallel. I chose the platform architecture to specifically support that as well as possible.

That's a start. If you want to, feel free to think along. If you still feel worried about it (probably, you should be), then help me find the biggest gaps, so we can talk about it.

### RE: budget

Just to clarify these budgets: Contrary to some of the examples that you mentioned (community forum, unofficial blog), **I have developed this platform without incurring any costs to the Coop as of yet**. Joshy supports funding for _creating new DevEd content_. This means that the stated costs are for creating content that can live on this platform, but could as well live on other platforms (for instance: the archdoc would need an update regardless, etc.).

To be fair: I'll still look for sponsorship on developing the platform itself, too. But that's because I believe it'll be valuable, and I'm committed to proving that. So there my personal incentives are aligned with addressing your primary concerns.

Does that help clarify?

What do you think of this distribution approach?

**Comment by JoshOrndorff** [comment]
Indeed I will support content creation related to educational materials, examples, architecture, etc. I don't know how to do that in a big issue like this without the possibility of an excited bounty hunter coming along and twisting my words into sponsoring the entire thing for an outrageous amount.

Just to make an on-the-record statement, I like what you're doing here and hope it gets worked into the website. That is beyond the scope of what I'm allowed to sponsor though.

@pmoorman Have you had any interaction with Kevin about this? I'd love this bounty to be sully sponsored and be synergistic with the "learning center" that we talk about sometimes or developer.rchain.coop. That would take a ton off of my plate and allow me to focus on creating content and leading learning groups and whatnot. Sounds much nicer than figuring out who to talk to and in what format to get content that has existed since before I started working for the coop into the website.

**Comment by dckc** [comment]
OK, while we're branstorming on scope, I'll put the flag back on...

**Comment by pmoorman** [comment]
Few thoughts regarding scope:

1. Maybe it's best to treat this as an "umbrella" issue with, without a budget, and reference other issues that have specific scopes and clear budgets. (e.g. the ArchDoc revamp has it's own budget & issue)
2. Doing this would also allow different pieces of content to be sponsored by different organizations. If Joshy can't sponsor governance content (for instance), maybe other people can. Different issues would support this better, from an accounting point-of-view
3. I believe specifically that content around "RChain in non-technical terms" (accessible for both devs and DApps decision-makers would fit Joshy's scope. I intent the "feature pages" to provide this. This specific bounty I could break out into a separate issue, as per point 1.)
4. As an aside: @patrickM727 reached out to request if I'd like to be a pm on the website, and this library effort. This is in the early stages, but could provide an alternative way to support the core of this effort. I don't know how it can support others like @TrenchFloat, though

My suggestion —for now, at least— is to swap the `SMART` label for the `Discussion` label...

**Comment by JoshOrndorff** [comment]
1,2. I love the idea of this being a discussion meta issue and other more focused bounties for content.
3. This is a huge synergy with our elearning initiative
4. You should do it. I also don't know how to support @TrenchFloat along with this offer it's confusing. That's why we need things like @glenbraun 's RHours. Luckily we're supporting a POC dapp for it. https://github.com/rchain/bounties/issues/1006

**Comment by pmoorman** [comment]
@JoshOrndorff understood!

For anyone who might read this and also wonders, as I did, _"what's this RHours thing?"_ Here's a link: https://github.com/RHours/RHours

point 3 => yeah, plan to start work on that in the next week or so. Will cross-reference the issues when I create them.
point 4.1 => I'll let everyone know how negotiation goes, in this issue.
point 4.2 => RE: trenchfloat et al. ...still something to figure out. The way I read it, at least Trenchfloat can help create DevEd content (and potentially other content that has sponsors), which seems good enough for now.

**Comment by dckc** [comment]
I tried out wikipedia today... didn't really come up with anything that I like.

https://info.rhobot.net/wiki/Main_Page
